File: 75-sort-colors/sort-colors.java
class Solution {
    public void sortColors(int[] nums) {
        int low = 0, high = nums.length - 1, current = 0;

        while(current <= high){
            if(nums[current] == 0){
                swap(nums, current, low);
                low++;
                current++;
            }else if(nums[current] == 2){
                swap(nums, current, high);
                high--;
            }else{
                current++;
            }
        }   
    }

    private void swap(int[] nums, int i, int j){
        int temp = nums[i];
        nums[i] = nums[j];
        nums[j] = temp;
    }
}

File: 75-sort-colors/sort-colors.py
class Solution(object):
    def sortColors(self, nums):
        """
        :type nums: List[int]
        :rtype: None
        Do not return anything, modify nums in-place instead.
        """

        low = 0
        current = 0
        high = len(nums) - 1

        while current <= high:

           
            if nums[current] == 0:
                nums[current], nums[low] = nums[low], nums[current]
                low += 1
                current += 1

           
            elif nums[current] == 2:
                nums[current], nums[high] = nums[high], nums[current]
                high -= 1


            else:
                current += 1